lib/events.ts

```diff
@@ -84,7 +84,7 @@ export async function trackCommand(commandSignature: string) {
 			try {
 				const balena = getBalenaSdk();
 				const $username = await balena.auth.whoami();
-				storage.set('cachedUsername', {
+				await storage.set('cachedUsername', {
```

Comment:
Note there are 3 commits in the PR - it may be helpful to review them separately.

Reply:
Ah.  It's a pity we don't have a check for these types of bug.  There are cases where a promise is returned and ignored, but it would be nice to have a way to formally specify that intention.